### PR TITLE
fix: issues from new audit report for gnfd contracts

### PR DESCRIPTION
**Pull request by cosinlink**
### Description

fix issues from new audit report for gnfd contracts

### Changes

Notable changes:
* [fix: [L02] Potential calculation error, add more check for block.chainid](https://github.com/bnb-chain/greenfield-contracts/commit/226cd6d19348b8cb42db61f4ef7cc8fb174951ff)
* [fix: NEW [Q02] Threshold may not be reasonable, set quorum to 2 for USPEND PROPOSAL](https://github.com/bnb-chain/greenfield-contracts/commit/12cdd57b3cd651353dd3d7b525f979eca343833c) 
* [fix: verified validators count must exceed 2/3 of total validators](https://github.com/bnb-chain/greenfield-contracts/commit/69a3c40caa9416bc77038d5184decfccad00e058)
* [fix: NEW [L01] Reentrancy is possible in retryPackage, add reentry lock for retryPackage](https://github.com/bnb-chain/greenfield-contracts/commit/5c6bfd4d850551feebfb70858c0d1737f4f499e1)
* [fix: NEW [Q01] Contract iterates over storage arrays, optimize relayer check on single cross-contract call](https://github.com/bnb-chain/greenfield-contracts/commit/f61c581cfd2ec79cfc8c7d7a78190021b40601ab)
* ...

### CODE REVIEW

**Review by Pythonberg1997 — APPROVED** [review]

File: contracts/CrossChain.sol [hunk 1]
```diff
@@ -209,12 +197,10 @@ contract CrossChain is Config, Initializable, ICrossChain {
         require(sequence == channelReceiveSequenceMap[channelId], "sequence not in order");
         channelReceiveSequenceMap[channelId]++;
 
-        // 2. check valid relayer
-        _checkValidRelayer(eventTime);
-
-        // 3. verify bls signature
+        // 2. check valid relayer And 3. verify bls signature
+        require(msg.sender == tx.origin, "only EOA");
```
Comment (guagualvcha) [inline contracts/CrossChain.sol:201]: can you make this as a modifier?

File: contracts/GnfdLightClient.sol [hunk 2]
```diff
@@ -218,9 +228,14 @@ contract GnfdLightClient is Initializable, Config, ILightClient {
 
         ptr = ptr + CHAIN_ID_LENGTH;
         uint256 valNum = (size - CONSENSUS_STATE_BASE_LENGTH) / VALIDATOR_BYTES_LENGTH;
+
+        for (uint256 idx = 0; idx < validatorSet.length; idx++) {
+            isRelayer[validatorSet[idx].relayerAddress] = false;
```
Comment (guagualvcha) [inline contracts/GnfdLightClient.sol:233]: use delete?

File: contracts/CrossChain.sol (hunk 1 above)
Comment (cosinlink) [reply contracts/CrossChain.sol]: fixed

File: contracts/GnfdLightClient.sol (hunk 2 above)
Comment (cosinlink) [reply contracts/GnfdLightClient.sol]: fixed

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**